### PR TITLE
Use GOOGLE_APPLICATION_CREDENTIALS and GCLOUD_PROJECT vars in RemoteGcsHelper

### LINKED ISSUE #335: Remove RemoteGcsHelper specific environment variables for setting projectId and key.
As discussed in #331 we should depend on the already existing `GOOGLE_APPLICATION_CREDENTIALS` and `GCLOUD_PROJECT` environment variables instead of the `RemoteGcsHelper` specific `GCLOUD_TESTS_KEY` and` GCLOUD_TESTS_PROJECT_ID` environment variables.

### PR DESCRIPTION
**Pull request by mziccard**
This fixes #335

### CODE REVIEW

File: gcloud-java-storage/src/main/java/com/google/gcloud/storage/testing/RemoteGcsHelper.java [hunk 1]
```diff
@@ -145,22 +145,17 @@ public static RemoteGcsHelper create(String projectId, String keyPath)
...
+   * environment variables: {@code GCLOUD_PROJECT} and {@code GOOGLE_APPLICATION_CREDENTIALS}.
    *
    * @return A {@code RemoteGcsHelper} object for the provided options.
    * @throws com.google.gcloud.storage.testing.RemoteGcsHelper.GcsHelperException if environment
-   *     variables {@code GCLOUD_TESTS_PROJECT_ID} and {@code GCLOUD_TESTS_KEY} are not set or if
-   *     the file pointed by {@code GCLOUD_TESTS_KEY} does not exist
+   *     variables {@code GCLOUD_PROJECT} and {@code GOOGLE_APPLICATION_CREDENTIALS} are not set or
+   *     if the file pointed by {@code GOOGLE_APPLICATION_CREDENTIALS} does not exist
    */
   public static RemoteGcsHelper create() throws GcsHelperException {
```
Comment (aozarov) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/testing/RemoteGcsHelper.java]: Explicit projetId and key verification is fine.

Another option would be to only document that this `create` method is based on the default lookup of [projectId](https://github.com/GoogleCloudPlatform/gcloud-java#specifying-a-project-id) and [credentials] (https://github.com/GoogleCloudPlatform/gcloud-java#authentication) which allows more more flexibility.

Is the current way your preference?

Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/testing/RemoteGcsHelper.java]: I thought about it and wasn't sure how to proceed.
I don't expect much users to run integration tests on top of GCE or AE (is it even possible?).

I don't have a preference here, if you prefer we can use default lookup, I'm happy to change.

Comment (aozarov) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/testing/RemoteGcsHelper.java]: > I don't expect much users to run integration tests on top of GCE or AE (is it even possible?).

I am pretty sure it is possible but I don't know how common it is.

> I don't have a preference here, if you prefer we can use default lookup, I'm happy to change.

My preference would be to use default lookup.

Comment (mziccard) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/testing/RemoteGcsHelper.java]: Let me put this on hold until we discuss and possibly close #341

Comment (aozarov) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/testing/RemoteGcsHelper.java]: I sent #343 to fix this issue

**Comment by mziccard** [comment]
I fixed the comment and now RemoteGcsHelper uses default auth credentials.
This should not be merged until we fix default auth credentials NPE.

**Comment by ajkannan** [comment]
#344 is now merged in, which fixes the default auth credentials NPE

**Comment by mziccard** [comment]
I realized now that using default authentication credentials causes signUrl method to fail (see [here](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-storage/src/main/java/com/google/gcloud/storage/StorageImpl.java#L587)) and consequently also signUrl IT tests fail.

We can manually pass a `ServiceAccountAuthCredentials` to `signUrl` but that is the same as explicitly creating `ServiceAccountAuthCredentials` from the env variables in `RemoteGcsHelper`.

I suggest that we do not use default lookup in `RemoteGcsHelper` unless you know a way to workaround the signUrl problem

File: gcloud-java-storage/src/main/java/com/google/gcloud/storage/testing/RemoteGcsHelper.java [hunk 2]
```diff
@@ -45,8 +45,8 @@
 
   private static final Logger log = Logger.getLogger(RemoteGcsHelper.class.getName());
   private static final String BUCKET_NAME_PREFIX = "gcloud-test-bucket-temp-";
-  private static final String PROJECT_ID_ENV_VAR = "GCLOUD_TESTS_PROJECT_ID";
-  private static final String PRIVATE_KEY_ENV_VAR = "GCLOUD_TESTS_KEY";
+  private static final String PROJECT_ID_ENV_VAR = "GCLOUD_PROJECT";
```
Comment (aozarov) [inline gcloud-java-storage/src/main/java/com/google/gcloud/storage/testing/RemoteGcsHelper.java]: These are not used anymore, right?

**Comment by aozarov** [comment]
> I realized now that using default authentication credentials causes signUrl method to fail (see here) and consequently also signUrl IT tests fail.

We could also support it when `GoogleCredentials` is instanceof [ServiceAccountCredentials](https://github.com/google/google-auth-library-java/blob/281d485e679545ed06f46e6090fc5d47f4a140e8/oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java). There are various ways we could deal with it.

**Comment by mziccard** [comment]
In 82ebc71 I fixed the comment and:
- Added a method to convert `ApplicationDefaultCredentials` to `ServiceAccountAuthCredentials` if `GoogleCredentials` object is an instance of `ServiceAccountCredentials`
- Add type check and conversion to `Storage.signUrl` to work with `ApplicationDefaultCredentials`

**Comment by aozarov** [comment]
Great! Looks good to me. Any reason you still have the "don't merge" label?

**Comment by mziccard** [comment]
I just forgot to remove :)

**Pull request merged**